src/storage/mod.rs

```diff
@@ -7396,6 +7416,99 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_scan_lock_with_memory_lock() {
+        let storage = TestStorageBuilderApiV1::new(MockLockManager::new())
+            .build()
```

Comment:
I just noticed that the test storage seems not enabling in-memory pessimistic locks by default.
How about testing both enabling and disabling?

Reply:
Thanks for the reminding, both are added and so is the memory lock checking, PTAL.